primitives/domain/src/lib.rs

```diff
@@ -187,11 +217,19 @@ pub const MAX_ASSET_NAME_LEN: u32 = 32;
 pub struct AssetName(pub BoundedVec<u8, ConstU32<MAX_ASSET_NAME_LEN>>);
 
 const MAINCHAIN_PUBLIC_KEY_LEN: usize = 32;
+const ETH_PUBLIC_KEY_LEN: usize = 32; // TODO ETH: is this correct length?
```

Comment:
Yes and no.
Story: historically this project started as Rust version of a Ethereum client that was able to observe Cardano. From this you can see SidechainPubKey. It is 33 bytes, because in this project we have always used the parity bit (hence all keys in hex start with 02 or 03). It is another complication for Minotaur, how to deal with existence of these keys in the project (which are user facing) and not make it very complicated for the user. Should already existing keys be used to get ETH stake address?

Reply:
From Minotaur point of view any key is an opaque type with some trait(s) as constraints (e.g. `Key: MainchainKey`).
From user point of view, I would prefer to have clear distinction between Cardano, Ethereum and PC keys/addresses so that it is clear where each type of key should be used. 
For example, if I want to run PC node and restake both my ADA and ETH, then I already have and understand the difference between Cardano and Ethereum keys/addresses. Then it is very naturally for me to learn about PC keys/addresses, how they look like and how to create and use them. 
If SidechainPubKey has Ethereum legacy, this is probably has to be fixed sooner or later as technical debt.

Reply:
I agree with fixing technical debt. I also do not want to add new one. I only try to raise awareness that:
* ECDSA public key can have different lengths
* For the "same lenght" one can one have version with and without a parity bit
* The parity bit is written as extra byte
* There are many ways of encoding a bit in a byte

This project already has one ECDSA public key in use. This public key is 33 bytes long. 32 bytes for a key and 1 byte that encodes parity bit as either 02 or 03.

This whole message is not trying me to forbid adding 32 bytes ECDSA public key w/o parity bit, I'm only saying that it will probably be confusing (added to already confusing Substrate layers over cryptography).

Reply:
Ah, that is important context which I didn't consider at all here. 
Then we can make it 33 bytes or even define it as
```rust 
pub struct EthPublicKey(pub Vec<u8>);
```
From Minotaur's perspective it is not important at all, so please let me know how it should be defined and I will change it.